### PR TITLE
Refactoring out the get_worst_possible_profit method

### PR DESCRIPTION
**Pull request by jsphon**
I'd find the separate win/lose exposures useful when controlling risk in my strategy code.

### CODE REVIEW

**Comment by jsphon** [comment]
Withdrawing for the time being, I think there might be some exposure problems that could be improved. I will investigate...

**Comment by jsphon** [comment]
Reopenning this PR.
The changes allow a LAY bet to hedge a BACK bet, reducing exposure.
Please see `test_validate_selection2a` and `test_validate_selection2b` for examples.

**Comment by liampauling** [comment]
With this being a breaking change to the return of a public method we would need to add some depreciation warnings first or add another method instead, thoughts?

**Comment by jsphon** [comment]
I don't think this will change the result of the public method `selection_exposure`, and none of the tests for that method have broken/changed as a result.

This change could potentially affect users who don't want backs and lays to offset each other, from a risk perspective. If those users need to be a consideration, perhaps we could add a flag to `trading_controls.py`, to let users decide whether or not to use exposure offsetting.

File: flumine/markets/blotter.py [hunk 1]
```diff
@@ -118,8 +127,15 @@ def selection_exposure(self, strategy, lookup: tuple) -> float:
...
             matched_exposure[1] + unmatched_exposure[1] + moc_lose_liability
         )
-        exposure = -min(worst_possible_profit_on_win, worst_possible_profit_on_lose)
-        return max(exposure, 0.0)
+
+        return {
+            "matched_profit_if_win": matched_exposure[0],
+            "matched_profit_if_lose": matched_exposure[1],
+            "worst_potential_unmatched_profit_if_win": unmatched_exposure[0],
+            "worst_potential_unmatched_profit_if_lose": unmatched_exposure[0],
```
Comment (liampauling) [inline flumine/markets/blotter.py:135]: Typo?

```
unmatched_exposure[0] -> unmatched_exposure[1]
```

File: flumine/markets/blotter.py [hunk 2]
```diff
@@ -77,6 +77,15 @@ def selection_exposure(self, strategy, lookup: tuple) -> float:
...
             zero = no potential loss
         """
+        exposures = self.get_exposures(strategy=strategy, lookup=lookup)
+        exposure = -min(
+            exposures["worst_possible_profit_on_win"],
+            exposures["worst_possible_profit_on_lose"],
+        )
+        return max(exposure, 0.0)
+
+    def get_exposures(self, strategy, lookup: tuple) -> dict:
```
Comment (liampauling) [inline flumine/markets/blotter.py:87]: Move unit tests to this function

File: flumine/markets/blotter.py [hunk 3]
```diff
@@ -77,6 +77,15 @@ def selection_exposure(self, strategy, lookup: tuple) -> float:
             positive = potential loss
             zero = no potential loss
         """
+        exposures = self.get_exposures(strategy=strategy, lookup=lookup)
```
Comment (liampauling) [inline flumine/markets/blotter.py:80]: Create new unit test for this function with get_exposures patched

**Review by liampauling — CHANGES_REQUESTED** [review]
Misread the git diff, few comments

**Comment by jsphon** [comment]
Thanks for looking at this Liam. Following your comments, I have updated the code.

**Review by liampauling — APPROVED** [review]

**Pull request merged**